### PR TITLE
Adds special attacks

### PR DESCRIPTION
**Pull request by KayeArray**
:cl: banthebantz
add: Added special attacks to certain weapons, just click the action icon and attack to activate
add: Special attacks cost stamina, make sure you read those descriptions carefully
/:cl:

[why]: Diversifies weapons a bit with unique attacks.

Currently the attacks are as follows:
Knives: Cost 30 stamina. Slash at arteries to cause heavy bleeding.

Butterfly Knives: Cost 15 stamina. Slice off a butt instantly.

Eswords: Cost 40 stamina. Throw yourself at an enemy from a tile away, impale and tear out their stomach. Risky because they have a small window to evade.

Dual eswords: Cost 50 stamina. Meme attack that makes you flail about at random rapidly attacking anything in range, lethal but very impractical.

Baseball bats: Cost 40 stamina. Normal attack that causes brain damage and confusion.

Stunbatons: Cost 35 stamina. Shock yourself and others with a tesla blast.

Chain of command: Cost 35 stamina. Trip and reel in fleeing criminal scum.

### CODE REVIEW

File: hippiestation/code/game/objects/items.dm [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/obj/item
+	var/special_attack = FALSE
+	var/special_name = "generic"
+	var/special_desc = "not supposed to see this"
+
+/obj/item/proc/do_special_attack(atom/target, mob/living/carbon/user, proximity_flag, cost)
+	if(user.getStaminaLoss() >= (100 - cost))
+		to_chat(user,"<span class='warning'> You don't have enough stamina to do this!")
```
Comment (ghost) [inline hippiestation/code/game/objects/items.dm]: Span class isn't terminated. Also a space when there shouldn't be.

Comment (KayeArray) [reply hippiestation/code/game/objects/items.dm]: How so?

Comment (KayeArray) [reply hippiestation/code/game/objects/items.dm]: nevermind I see it

**Comment by GuyonBroadway** [comment]
>Slice off a butt instantly

Well I've found my new clown gimmick.

**Comment by Arzion5** [comment]
https://www.youtube.com/watch?v=R3oSAVwcD4U
+1

File: hippiestation/code/game/objects/items/weapons/special_attacks.dm [hunk 2]
```diff
@@ -0,0 +1,167 @@
+//when making a weapon with a special attack or adding a special attack, please add the special vars and proc here for the sake of organisation
+/obj/item/kitchen/knife
+	special_name = "Artery slash"
+	special_desc = "COST: 30 STAMINA. A precision slash targeting an artery, causes extra bleeding."
+	actions_types = list(/datum/action/item_action/special_attack)
+
+/obj/item/kitchen/knife/do_special_attack(atom/target, mob/living/carbon/user, proximity_flag, cost = 30)
+	..()
```
Comment (ghost) [inline hippiestation/code/game/objects/items/weapons/special_attacks.dm]: ..() is used twice

File: hippiestation/code/game/objects/items/weapons/special_attacks.dm [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		user.adjustStaminaLoss(cost)
+		src.special_attack = FALSE
+		H.bleed_rate = min(H.bleed_rate + 10, 10)
+		H.blood_volume -= 25
+
+		for(var/turf/T in range(H.loc, 1))
+			H.add_splatter_floor(T)
+		H.visible_message("<span class='danger'>[user] slashes open one of [H]'s arteries with [src]!</span>", "<span class='userdanger'>[user] slices open one of your arteries with [src]!</span>")
+		playsound(H, 'sound/effects/splat.ogg', 50, 1)
+	..()
```
Comment (ghost) [inline hippiestation/code/game/objects/items/weapons/special_attacks.dm]: ..() is used twice

**Comment by PopNotes** [comment]
this is fucking fantastic thank you

**Comment by DinduStuffin** [comment]
this is an amazing idea holy shit
we need more PRs like this

I can't wait to see gangs of people running around hunting butts
+1

**Comment by PopNotes** [comment]
As funny as the butt slicing is, I think it'd be best if butterfly knives had, say, a slowing effect on successful special attack backstab.

**Comment by GuyonBroadway** [comment]
Nah the slowing effect would just make balisong boning too easy. 

Butt slicing is a fun gimmick that promotes meme gameplay which is good.

**Comment by PopNotes** [comment]
Butt slicing doesn't actually do anything to anyone and has no effect on people who already don't have asses. Give it to the honk render or something.

**Comment by ghost** [comment]
would it not be better to make a new function that calls the pre-setup, attack itself and then the clean up and just have it so each object modifies that middle attack proc?

**Comment by KayeArray** [comment]
done

**Comment by ghost** [comment]
looks a lot nicer now

File: hippiestation/code/game/objects/items/weapons/special_attacks.dm [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+	special_name = "Dance of the Fates"
+	special_desc = "COST: 50 STAMINA. Launch into an uncontrollable flurry of attacks slicing up any enemies in range"
+	special_cost = 50
+	actions_types = list(/datum/action/item_action/special_attack)
+
+/obj/item/twohanded/dualsaber/do_special_attack(atom/target, mob/living/carbon/user)
+	if(wielded)
+		user.visible_message("<span class='danger'>[user] begins to flail around wildly!</span>")
+		user.confused += 200
+		src.block_chance = 100
```
Comment (ghost) [inline hippiestation/code/game/objects/items/weapons/special_attacks.dm:79]: shouldn't this be reset?

**Comment by PopNotes** [comment]
Special attacks should be accompanied with some kind of cool sound effect, I'm thinking. What should it beee.

**Comment by KayeArray** [comment]
up to you

**Comment by mlgdatboi** [comment]
do some spartan kick that gibs them and it goes BWAAAAAP

**Comment by ghost** [comment]
i like them all bar the tesla one

station engineer antag breaks into sec locker room from spess and steals a bunch of stun batons and now instantly stuns everyone in the room by turning the thing on and hitting the floor

**Comment by KayeArray** [comment]
it's been nerfed

**Comment by GuyonBroadway** [comment]
Thank god, that was the best/worst thing ever. As the CMO I could jsut wear my medical hardsuit and go all wrath of Olympus on everyones backsides with no repercussions.

**Comment by PopNotes** [comment]
I want the officer's sabre to have a special move that ignores shields and does 50% armor penetration for that attack only.

**Comment by KayeArray** [comment]
🤔

**Comment by KayeArray** [comment]
fixing

**Comment by PopNotes** [comment]
wut

**Pull request closed without merging**